lib/elixir/lib/module/locals_tracker.ex

```diff
@@ -98,6 +98,20 @@ defmodule Module.LocalsTracker do
     {unreachable(reachable, reattached, private), collect_warnings(reachable, private)}
   end
 
+  @doc """
+  Collect undefined functions based on local calls and existing definitions\][=]
+  """
+  def collect_undefined_locals({_set, bag}, all_defined) do
+    undefined =
+      Enum.reduce(all_defined, %{}, fn {pair, _, _, _}, acc ->
+        undefined_from(bag, pair, acc)
```

Comment:
I think you can simplify this by passing the `set` table to undefined_from and using `:ets.pmember(set, {:def, {name, arity}})` to check if the function exists. This way you dont need a map mixed with ok and undefined function, but you can just build the list directly, something like:

```elixir
for {pair, _, _, _} <- all_defined,
    {local, line} <- out_neighbours(pair),
    not :ets.member(set, local),
    do: {get_meta(line), {:undefined_function, local}}
```